networkit/cpp/graph/GraphTools.cpp

```diff
@@ -73,6 +75,55 @@ Graph restoreGraph(const std::vector<node>& invertedIdMap, const Graph& G) {
 	return Goriginal;
 }
 
+Graph extractLargestConnectedComponent(const Graph &G, bool compactGraph) {
+	if (!G.numberOfNodes())
+		return Graph(G);
+
+	ConnectedComponents cc(G);
+	cc.run();
+
+	auto compSizes = cc.getComponentSizes();
+	auto largestCC = std::max_element(
+		compSizes.begin(), compSizes.end(),
+		[](const std::pair<index, count> &x, const std::pair<index, count> &y) {
+			return x.second < y.second;
+		});
+	auto components = cc.getComponents();
+	std::unordered_map<node, node> continuousNodeIds;
+	std::vector<bool> toRemove(G.upperNodeIdBound(), false);
+	count toRemoveCount = 0;
+
+	index nextId = 0;
+	G.forNodes([&](node u) {
+		if (cc.componentOfNode(u) == largestCC->first) {
+			continuousNodeIds[u] = nextId++;
+		}
+		else {
+			toRemove[u] = true;
```

Comment:
`toRemove` seems unnecessary. Why don't we just check `componentOfNode` again in the second loop?